pkg/detection/detect.go

```diff
@@ -83,13 +83,14 @@ func init() {
 				return false
 			}
 
-			data, err := io.ReadAll(r)
-			if err != nil {
+			var buf bytes.Buffer
+			if _, err := io.Copy(&buf, r); err != nil {
 				return false
 			}
+			data := buf.Bytes()
 
 			contents := make(map[string]interface{})
-			err = json.Unmarshal(data, &contents)
+			err := json.Unmarshal(data, &contents)
```

Comment:
Seems like `json.NewDecoder(r).Decode(&contents)` is better, no?

Reply:
I implemented this here https://github.com/aquasecurity/defsec/pull/1219/commits/06e77bdd7445030d1fd083f7805bdfb59f2aa839

results: https://github.com/simar7/copy-bench/blob/main/results.md#iocopy--bufbytes--jsonunmarshal-vs-jsonnewdecoder
tests: https://github.com/simar7/copy-bench/blob/main/main.go#L45-L79